contracts/exchange/contracts/src/MixinExchangeCore.sol

```diff
@@ -177,6 +171,66 @@ contract MixinExchangeCore is
         return orderInfo;
     }
 
+    /// @dev Validates context for fillOrder. Succeeds or throws.
+    /// @param order to be filled.
+    /// @param orderInfo OrderStatus, orderHash, and amount already filled of order.
+    /// @param takerAssetFillAmount Desired amount of order to fill by taker.
+    /// @param takerAssetFilledAmount Amount of takerAsset that will be filled.
+    /// @param makerAssetFilledAmount Amount of makerAsset that will be transfered.
+    function assertValidFill(
+        Order memory order,
+        OrderInfo memory orderInfo,
+        uint256 takerAssetFillAmount,  // TODO: use FillResults
+        uint256 takerAssetFilledAmount,
+        uint256 makerAssetFilledAmount
+    )
+        public
```

Comment:
Is this useful as a public function? Feels like it might not be, since you need to know the results fo the fill before calling it.

Reply:
I agree. There's a lot of burden to get to the point to where you could even make this call. I think a higher level function would be more useful (maybe there's one already exposed). Unless @fabioberger has any objections, I'm making this internal again.